src/json.c

```diff
@@ -510,21 +512,51 @@ int json_nvmet_view_exports(struct NVMET_PROFILE *nvmet, struct NVMET_PORTS *por
 		json_object_set_new(object, "namespace", json_integer(nvmet->namespc));
 		json_object_set_new(object, "device", json_string(nvmet->device));
 		json_object_set_new(object, "enabled", json_string((nvmet->enabled == ENABLED) ? "true" : "false"));
+		json_t *hosts_allowed_array = json_array();
+		allowed_hosts = nvmet->allowed_hosts;
+		while (allowed_hosts != NULL) {
+			json_t *object_h = json_object();
+			json_object_set_new(object_h, "host", json_string(allowed_hosts->allowed_host));
+			json_array_append_new(hosts_allowed_array, object_h);
+			allowed_hosts = allowed_hosts->next;
+		}
+		json_object_set_new(object, "allowed_hosts", hosts_allowed_array);
+
+		json_t *assigned_ports_array = json_array();
+		ports = nvmet->assigned_ports;
+		while (ports != NULL) {
+			json_t *object_p = json_object();
+			json_object_set_new(object_p, "port", json_integer(ports->port));
+			json_object_set_new(object_p, "address", json_string(ports->addr));
+			json_object_set_new(object_p, "protocol", json_string(ports->protocol));
+			json_object_set_new(object_p, "nqn", json_string(ports->nqn));
+			json_array_append_new(assigned_ports_array, object_p);
+			ports = ports->next;
+		}
+		json_object_set_new(object, "assigned_ports", assigned_ports_array);
+
 		json_array_append_new(nvmet_array, object);
 		nvmet = nvmet->next;
 	}
 	json_object_set_new(nvmet_object, "nvmet_targets", nvmet_array);
 	json_array_append_new(array, nvmet_object);
 
-	while (ports != NULL) {
-		json_t *object = json_object();
-		json_object_set_new(object, "port", json_integer(ports->port));
-		json_object_set_new(object, "address", json_string(ports->addr));
-		json_object_set_new(object, "protocol", json_string(ports->protocol));
-		json_object_set_new(object, "nqn", json_string(ports->nqn));
-		json_array_append_new(ports_array, object);
-		ports = ports->next;
+	nvmet = nvmet_orig;
```

Comment:
Since this is being handled above, it seems a bit redundant here below. No worries. I can double-check after I merge.